lib/elixir/lib/task/supervised.ex

```diff
@@ -10,26 +12,37 @@ defmodule Task.Supervised do
   end
 
   def start_link(caller, link, info, fun) do
-    :proc_lib.start_link(__MODULE__, :reply, [caller, link, info, fun])
+    {:ok, spawn_link(caller, link, info, fun)}
   end
 
-  def async(caller, info, mfa) do
-    initial_call(mfa)
-    ref = receive do: ({^caller, ref} -> ref)
-    send caller, {ref, do_apply(info, mfa)}
+  def spawn_link(caller, link \\ :nolink, info, fun) do
+    :proc_lib.spawn_link(__MODULE__, :reply, [caller, link, info, fun])
```

Comment:
We would need to do exactly the same error handling if we didn't use `:proc_lib` because the emulator does not send a log message when a process exits due to `exit/1`, only on uncaught `error` or `throw`. For example a failed `GenServer.call/1` calls `exit/1`, which would not trigger the emulator log message when using `spawn_link` without `:proc_lib`. Also using a plain `spawn_link` we would not get the `caller` or `mfa` in the logs.

`:proc_lib` provides its own error handling: a `crash_report`. This provides a lot of information, though much of it would only be useful when debugging. However it does provide information about the neighbours. Neighbours are linked processes that are not trapping exits. These processes will get the exit signal of the terminating process and be abruptly halted and so unable to send their own log messages. In the case of `Task.async` and `Task.Supervisor.async` this effectively produces a nested `crash_report` of the `caller` if it is not trapping exits. `caller` may not be the only neighbour as the `caller` could have links too. This means details of the whole error layer are logged.